src/Haskell/Ide/Engine/Plugin/HieExtras.hs

```diff
@@ -537,6 +539,101 @@ getModule df n = do
   return (pkg, T.pack $ moduleNameString $ moduleName m)
 
 -- ---------------------------------------------------------------------
+-- TODO: there has to be a simpler way, using the appropriate GHC internals
+findIdForName :: GHC.TypecheckedModule -> GHC.Name -> IdeM (Maybe GHC.Id)
+findIdForName tm n = do
+  let t = GHC.tm_typechecked_source tm
+  let r = SYB.something (SYB.mkQ Nothing worker) t
+      worker (i :: GHC.Id) | nameUnique n == varUnique i = Just i
+      worker _ = Nothing
+  return r
+
+-- ---------------------------------------------------------------------
+
+getTypeForName' :: GHC.TypecheckedModule -> GHC.Name -> IdeM (Maybe GHC.Type)
+getTypeForName' tm n = do
+  mId <- findIdForName tm n
+  case mId of
+    Nothing -> return Nothing
+    Just i  -> return $ Just (varType i)
+
+-- | Return the type definition
+findTypeDef :: Uri -> Position -> IdeDeferM (IdeResult [Location])
+findTypeDef uri pos = pluginGetFile "findTypeDef: " uri $ \file -> do
```

Comment:
I’m finding this function very difficult to read due to the very short variable names and all the branches the logic goes to. Would it be possible to prettify it a little? Let me know if you need help with this

Reply:
The functions are actually copies of HaRe and typeDef. But you are right, I will spent time on improving naming and documentation
It took, roughly 30 hours to write these lines of code, more documentation is needed!

Reply:
Is it better now?